src/main/python/pgshovel/replication/streams/kafka.py

```diff
@@ -40,12 +54,44 @@ def consume(self, state):
             consumer.seek(delta, 1)
             assert consumer.offsets[0] == offset
 
-        for offset, message in consumer:
-            decoded = self.codec.decode(message.value)
-            state = validate_state(state, offset, decoded)
+        for offset, message in self.__primed(consumer, state):
+            state = validate_state(state, offset, message)
             # XXX: This is necessary because of a bug in protocol buffer oneof.
             state = type(state).FromString(state.SerializeToString())
-            yield state, offset, decoded
+            yield state, offset, message
+
+    def __primed(self, consumer, state, attempts=5):
```

Comment:
Personally, I'd move this function off the class since I don't think there's is a good reason to override it, it doesn't use any instance attributes, and will make #28 easier in the future. (Honestly, I'd just move it to `pgshovel.replication.streams.utilities` or similar to facilitate the transition later and prevent it from being anything Kafka specific.)